client/finality-grandpa/src/import.rs

```diff
@@ -418,6 +423,80 @@ where
 
 		Ok(PendingSetChanges { just_in_case, applied_changes, do_pause })
 	}
+
+	/// Read current set id form a given state.
+	fn current_set_id(&self, id: &BlockId<Block>) -> Result<SetId, ConsensusError> {
+		match self.inner.runtime_api().current_set_id(&id) {
+			Ok(set_id) => Ok(set_id),
+			Err(sp_api::ApiError::Application(_)) => {
+				// The new API is not supported in this runtime. Try reading directly from storage.
+				// This code may be removed once warp sync to an old runtime is no longer needed.
+				for prefix in ["GrandpaFinality", "Grandpa"] {
```

Comment:
> Any substrate user that has not named it `GrandpaFinality`

Or "Grandpa".

At least it will work for Polkadot and Kusama. We will remove it in the next release.